web/packages/teleterm/src/ui/AccessRequestCheckout/useAccessRequestCheckout.test.tsx

```diff
@@ -230,83 +230,86 @@ test('after creating an access request, pending requests and specifiable fields
     wrapper,
   });
 
-  await waitFor(() => {
-    result.current.setSelectedReviewers([{ value: 'bob', label: 'bob' }]);
-    expect(result.current.selectedReviewers).toEqual([
-      { value: 'bob', label: 'bob' },
-    ]);
-  });
+  await act(() =>
```

Comment:
`await` on `act` is not necessary, unless the callback passed to `act` returns a promise. This is the case for a lot of `await act` in this file. TypeScript's lang server should highlight all of them if you have it enabled in your editor. There's also the `@typescript-eslint/await-thenable` rule, but enabling it requires adjusting the ESLint config a little bit.

Reply:
hrm, i actually do need them in almost all cases, b/c most are react state setters which are async? (the test fails without them)

Reply:
React state setters are technically sync and they don't need to be awaited. TS lang server highlights those callsites because `act` is given a sync function, so there's nothing to await.

The errors after removing `await` stem from the fact that `useAccessRequestCheckout` has an effect which fetches resource roles every time you change `pendingAccessRequestRequest`. This means each call to `updateNamespacesForKubeCluster` causes this effect to be run again.

At first I removed those await calls and added `waitFor` for `fetchResourceRolesAttempt`. But then I looked up [the docs for `act`](https://react.dev/reference/react/act#reference) and the React team actually recommends passing an async function to `act`. So I left the awaits, but I made sure that we always pass an async function there. This way the TS language server doesn't complain about those callsites.

Reply:
I made a PR with those changes: https://github.com/gravitational/teleport/pull/48969